resolver/type_syntax/type_syntax.cc

```diff
@@ -1017,9 +1019,17 @@ optional<TypeSyntax::ResultType> getResultTypeAndBindWithSelfTypeParamsImpl(core
             auto klass = sym.asClassOrModuleRef();
             // the T::Type generics internally have a typeArity of 0, so this allows us to check against them in the
             // same way that we check against types like `Array`
-            if (klass.isBuiltinGenericForwarder() || klass.data(ctx)->typeArity(ctx) > 0) {
-                auto level = klass.isLegacyStdlibGeneric() ? core::errors::Resolver::GenericClassWithoutTypeArgsStdlib
-                                                           : core::errors::Resolver::GenericClassWithoutTypeArgs;
+            //
+            // TODO(jez) After T::Class change: fix the payload, fix all the codebases, and remove this check.
+            // (Leaving at least one version in between, so that there is a published version that
+            // supports both `Class` and `T::Class` as valid syntax.)
+            if (klass != core::Symbols::Class() &&
+                (klass.isBuiltinGenericForwarder() || klass.data(ctx)->typeArity(ctx) > 0)) {
+                // Class is not isLegacyStdlibGeneric (because its type members don't default to T.untyped),
+                // but we want to report this syntax error at `# typed: strict` like other stdlib classes.
+                auto level = klass.isLegacyStdlibGeneric() || klass == core::Symbols::Class()
+                                 ? core::errors::Resolver::GenericClassWithoutTypeArgsStdlib
+                                 : core::errors::Resolver::GenericClassWithoutTypeArgs;
```

Comment:
This should let users incrementally migrate, so there will be at least one version published that allows `T::Class` but does not require it.